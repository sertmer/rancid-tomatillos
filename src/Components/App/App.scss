@import '../../scss/mixins';
@import '../../scss/variables';

.Nav-div {
  display: flex;
  justify-content: center;
  flex-direction: row;
  align-items: center;
  height: 10%;
}

body {
  background: rgb(54,18,91);
  background: linear-gradient(
    0deg,
  rgba(54,18,91,1) 0%,
  rgba(60,24,128,1) 53%,
  rgba(98,26,185,1) 100%
  );
  height: 120vh;
}

main {
  display: flex;
  overflow: auto;
  height: 100vh;
  margin: 0;
}