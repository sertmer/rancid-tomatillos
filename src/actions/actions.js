export const getMovies = movies => ({
  type: 'GET_MOVIES',
  movies
});

export const currentUser = user => ({
  type: 'SET_CURRENT_USER',
  name: user.user.name,
  id: user.user.id,
  ratings: []
});

export const loggedIn = loggedIn => ({
  type: 'LOG_IN',
  loggedIn
});

export const getCurrentMovie = movie => ({
  type: 'GET_MOVIE',
  movie
});

export const getRatings = ratings => ({
  type: 'GET_RATINGS',
  ratings
});

