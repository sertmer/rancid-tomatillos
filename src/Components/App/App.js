import React, { Component } from 'react';
import { getMovies } from '../../actions/actions';
import MoviesContainer from '../MoviesContainer/MoviesContainer';
import './App.scss';
import Nav from '../Nav/Nav';
import Login from '../Login/Login'
import { connect } from 'react-redux';
import { Route } from 'react-router-dom';
import MovieShowPage from '../MovieShowPage/MovieShowPage';

class App extends Component {
  constructor() {
    super()
  }

  componentDidMount() {
    fetch('https://rancid-tomatillos.herokuapp.com/api/v1/movies')
      .then(res => res.json())
      .then(data => this.props.getMovies(data.movies))
      .catch(err => console.log(err))
  }

  render() {
    return (
      <body>
        <Route path='/' render={() =>
          <Nav />
          }
        />
        <Route exact path='/' render={() =>
          <main>
            <MoviesContainer />
          </main>
          }
        />
        <Route path='/login' render={() =>
          <Login />
          }
        />
        <Route path='/movies/:id' render={() =>
            <MovieShowPage />
          }
        />
      </body>
    )
  }
}

const mapDispatch = dispatch => ({
  getMovies: movies => dispatch(getMovies(movies))
})

export default connect(null, mapDispatch)(App)
