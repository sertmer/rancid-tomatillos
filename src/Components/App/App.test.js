import React from 'react';
import { shallow } from 'enzyme';
import { mapDispatch, App } from './App';
import { getMovies } from '../../actions/actions';


describe('App', () => {
  let wrapper;

  it('should match the snapshot', () => {
    wrapper = shallow(<App />)

    expect(wrapper.debug()).toMatchSnapshot()
  })
})
