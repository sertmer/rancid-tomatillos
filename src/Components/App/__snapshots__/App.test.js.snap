// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
"<body>
  <Route path=\\"/\\" render={[Function: render]} />
  <Route exact={true} path=\\"/\\" render={[Function: render]} />
  <Route path=\\"/login\\" render={[Function: render]} />
  <Route path=\\"/movies/:id\\" render={[Function: render]} />
</body>"
`;
